src/Client/DurableStorage/FilesystemDurableStorage.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+
+namespace Tuf\Client\DurableStorage;
+
+/**
+ * Class FilesystemLocalState
+ *
+ * A simple implementation of LocalStateInterface using the filesystem.
+ * Applications might want to provide an alternative implementation with better performance and error handling.
+ */
+class FilesystemDurableStorage implements \ArrayAccess
+{
+    /**
+     * @var string $rootPath
+     *   The filesystem directory where durable state is rooted.
+     */
+    protected $rootPath;
+
+    public function __construct(string $rootPath)
+    {
+        if (! is_dir($rootPath)) {
+            throw new \RuntimeException("Cannot initialize filesystem local state: \"$rootPath\" is not a directory.");
+        }
+
+        $this->rootPath = $rootPath;
+    }
+
+    protected function offsetToPath($offset)
```

Comment:
I disagree on this because the \ArrayAccess interface we're implementing uses the term "offset" and the purpose of this method is to translate between array offsets and filenames. Care 2/10.

Reply:
I mean that we should get rid of the term "offset" entirely because it's confusing/meaningless.